services/web/server/src/simcore_service_webserver/scicrunch/service_client.py

```diff
@@ -0,0 +1,189 @@
+"""
+    Client to communicate with scicrunch service API (https://scicrunch.org/api/)
+"""
+
+import logging
+import re
+from enum import IntEnum
+from http import HTTPStatus
+from typing import Any, List, MutableMapping, Optional
+
+import aiohttp
+from aiohttp import ClientSession, web
+from pydantic import ValidationError
+from servicelib.client_session import get_client_session
+from yarl import URL
+
+from ._config import RRID_PATTERN, SciCrunchSettings
+from .scicrunch_models import ListOfResourceHits, ResourceView
+
+logger = logging.getLogger(__name__)
+
+
+async def get_all_versions(
+    rrid: str, client: ClientSession, settings: SciCrunchSettings
+) -> List:
+    async with client.get(
+        f"{settings.api_base_url}/resource/versions/all/{rrid}",
+        params={"key": settings.api_key.get_secret_value()},
+        raise_for_status=True,
+    ) as resp:
+        body = await resp.json()
+        return body.get("data") if body.get("success") else []
+
+
+async def get_resource_fields(
+    rrid: str, client: ClientSession, settings: SciCrunchSettings
+) -> ResourceView:
+    async with client.get(
+        f"{settings.api_base_url}/resource/fields/view/{rrid}",
+        params={"key": settings.api_key.get_secret_value()},
+        raise_for_status=True,
+    ) as resp:
+        body = await resp.json()
+
+        assert body.get("success")  # nosec
+        return ResourceView(**body.get("data", {}))
+
+
+async def autocomplete_by_name(
+    guess_name: str, client: ClientSession, settings: SciCrunchSettings
+) -> ListOfResourceHits:
+    async with client.get(
+        f"{settings.api_base_url}/resource/fields/autocomplete",
+        params={
+            "key": settings.api_key.get_secret_value(),
+            "field": "Resource Name",
+            "value": guess_name.strip(),
+        },
+        raise_for_status=True,
+    ) as resp:
+        body = await resp.json()
+        assert body.get("success")  # nosec
+        return ListOfResourceHits.parse_obj(body.get("data", []))
+
+
+class ValidationResult(IntEnum):
+    UNKNOWN = -1
+    INVALID = 0
+    VALID = 1
```

Comment:
Please see below where used.

```suggestion
class ValidationResult(IntEnum):
    UNKNOWN = -1
    INVALID = 0
    VALID = 1

    @classmethod
    def format(cls, is_valid: bool) -> "ValidationResult":
        return cls.VALID if is_valid else cls.INVALID
```

Reply:
format for me denotes a string ...